dynomite/examples/local.rs

```diff
@@ -25,7 +25,7 @@ use uuid::Uuid;
 pub struct Book {
     #[dynomite(partition_key)]
     id: Uuid,
-    #[dynomite(rename = "bookTitle")]
+    #[dynomite(rename = "bookTitle", default)]
```

Comment:
need to test this. not quite working yet

Reply:
Doug, this will make the structures very verbose and will need a default implementation for any complex type. Would it better to assume that if a field can be absent it would be defined in the struct as Option<T> and we can just return None?

Reply:
Actually, Option<T> has None for default. So my only question here is if we can return None in all cases and avoid using #[dynomite(default)] altogether ...

Reply:
To answer your questions in order

1) I'd err on the side of being explicit for two practical reasons
 * I don't want the default behavior to hide potential application bugs
* Not all types that could impl Attribute are guaranteed to have Default impls. I could add that type constraint but I don't believe that's helpful in cases where you might want to impl attribute for a type that can't have a Default impl by design. These interfaces should be made to fit the types you have as a design goal.

2) procmacros aren't type-aware, only syntax-aware. There's not a clear way in a procmacro to dicern types and Rust intentionally avoids runtime reflection by not providing a convenient runtime reflection api. I don't think it's possible to to do this only for option types and relating to question one: I'd strongly prefer an explicit approach to avoid hiding application bugs. Application developers should know which fields should be present and which might not.

Reply:
Third thought I had. It might be useful in a future change to allow for a custom defaulting fn when a field is absent. `Default::default()` is just an mvp for a more common case.